upf/upf_api.c

```diff
@@ -1061,8 +1061,49 @@ vl_api_upf_tdf_ul_table_add_t_handler (vl_api_upf_tdf_ul_table_add_t * mp)
   REPLY_MACRO (VL_API_UPF_TDF_UL_TABLE_ADD_REPLY);
 }
 
-#include <upf/upf.api.c>
+/* API message handler */
+static void
+vl_api_upf_nat_pool_add_t_handler (vl_api_upf_nat_pool_add_t * mp)
+{
+  int rv = 0;
+  upf_main_t *sm = &upf_main;
+  vl_api_upf_nat_pool_add_reply_t *rmp = NULL;
+
+  u8 *name_vec = 0;
+  u8 *nwi_vec = 0;
+  u32 name_vec_len = 0;
+  ip4_address_t start, end;
+
+  mp->min_port = ntohs (mp->min_port);
+  mp->max_port = ntohs (mp->max_port);
+  mp->block_size = ntohl (mp->block_size);
+  do
+    {
+      if (mp->nwi_len > 64 || mp->name_len > 64)
+	{
+	  rv = VNET_API_ERROR_INVALID_VALUE;
+	  break;
```

Comment:
`goto` is often frowned upon, but not in VPP code. So in order to be consistent with the rest of the code, we either need to switch to this `do { ... } while(0)` construct elsewhere, or use `goto` here like in other API handlers (granted, some of them use `out` and some of them use `reply` label, but that IMO is a lesser inconsistency, even though also worth fixing at some point, maybe).
So I'd just use a `reply:` label and a `goto` for this case